force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,16 +139,21 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
-
+        String concatNames = '';
         // Loop through the list of scientists' names
+        for(String str : scientistNames){
 
             // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
+            If(str.contains('g') || str.contains('G')){
+                // Add the name to the result string, followed by a comma
+                concatNames += str + ',';
+                }
+            
+            }
         // Remove the last comma from the result string
+        String trimmedString = concatNames.removeEnd(',');
 
-        return null; // Replace null with the variable you used to store the result
+        return trimmedString; // Replace null with the variable you used to store the result
```

Comment:
Good,  rather than conditional logic around case sensitivity, I often prefer to just upper or lowercase the entire thing. 

```
  public static String findTheScientist() {
        // The list of scientists' names
        List<String> scientistNames = new List<String>{
                'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'
        };
        String result = '';
        for (String name : scientistNames) {
            if (name.toLowerCase().contains('g')) {
                result += name + ',';
            }
        }
        return result.removeEnd(',');
    }
    ```